### PR TITLE
Auto-Expand Nomai Text feature

### PR DESCRIPTION
**Pull request by GameWyrm**
Expands all Nomai text in the game by default when the option in the config file is turned on.

### CODE REVIEW

File: mod/NomaiTextQoL.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                __instance.ShowImmediate();
+            }
+        }
+
+        // fixes for the text not becoming properly grey when read
+        [HarmonyReversePatch, HarmonyPatch(typeof(NomaiText), nameof(NomaiText.SetAsTranslated))]
+        public static void base_SetAsTranslated(NomaiText instance, int id) { }
+
+        [HarmonyPrefix, HarmonyPatch(typeof(NomaiWallText), nameof(NomaiWallText.SetAsTranslated))]
+        public static bool NomaiWallText_SetAsTranslated_Transpiler(NomaiWallText __instance, ref int id)
```
Comment (Ixrec) [inline mod/NomaiTextQoL.cs:33]: is "_Transpiler" out of date?

**Comment by Ixrec** [comment]
Just so we don't forget something, issues I found testing this in-game:
- Enabling the setting mid-loop does not show any hidden text lines, and translating a line of Nomai text no longer reveals its child lines
- Text walls connected to projection stone readers appear to be showing multiple projection stones' worth of text lines
- The "Patch in..." log is too noisy

File: mod/NomaiTextQoL.cs (hunk 1 above)
Comment (GameWyrm) [reply mod/NomaiTextQoL.cs]: yep, fixed

File: mod/default-config.json [hunk 2]
```diff
@@ -1,56 +1,62 @@
...
+        "Separator - Crash Testing": {
+            "type": "separator"
+        },
+
+        "[DEBUG] Disable In-Game Console": false,
+        "[DEBUG] Don't Send Locations In-Game": false,
+        "[DEBUG] Don't Receive Items In-Game": false,
+        "[DEBUG] Don't Write To Save File In-Game": false
+    }
 }
```
Comment (Ixrec) [inline mod/default-config.json:62]: was the indentation change intentional? not that I have a strong opinion here, but the inconsistency makes it look like an overzealous autoformatter

File: mod/APRandomizer.cs [hunk 3]
```diff
@@ -331,6 +331,12 @@ private void Start()
             Victory.OnCompleteSceneLoad(scene, loadScene);
         };
 
+        // update the Nomai text setting before any can be created
+        LoadManager.OnStartSceneLoad += (scene, loadScene) =>
+        {
+            NomaiTextQoL.AutoNomaiText = AutoNomaiText;
+        };
```
Comment (Ixrec) [inline mod/APRandomizer.cs:338]: It has to be _start_ scene load, not the usual complete scene load?

File: mod/NomaiTextQoL.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                        {
+                            __instance._idToNodeDict[id].Value.SetTranslatedState(false);
+                            if (!NomaiWallText.s_revealAllAtStart)
+                            {
+                                for (int i = 0; i < owtreeNode.Children.Count; i++)
+                                {
+                                    revealedChildren = true;
+                                }
+                            }
+                        }
```
Comment (Ixrec) [inline mod/NomaiTextQoL.cs:86]: I assume something in here is different from the vanilla code, could you comment exactly where the difference is supposed to be?

File: mod/default-config.json (hunk 2 above)
Comment (GameWyrm) [reply mod/default-config.json:62]: Was likely just VS doing it, I didn’t manually change any formatting

File: mod/APRandomizer.cs (hunk 3 above)
Comment (GameWyrm) [reply mod/APRandomizer.cs:338]: Just want to be absolutely sure that no Nomai Wall Text exists yet.

File: mod/NomaiTextQoL.cs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+        // fixes for the text not becoming properly grey when read
+        [HarmonyReversePatch, HarmonyPatch(typeof(NomaiText), nameof(NomaiText.SetAsTranslated))]
+        public static void base_SetAsTranslated(NomaiText instance, int id) { }
+
+        [HarmonyPrefix, HarmonyPatch(typeof(NomaiWallText), nameof(NomaiWallText.SetAsTranslated))]
+        public static bool NomaiWallText_SetAsTranslated_Prefix(NomaiWallText __instance, ref int id)
+        {
+            // This code is copied from the base game and overrides the original method
+            if (!AutoNomaiText) return true;
```
[the remark this answers is not in the document]
Comment (GameWyrm) [reply mod/NomaiTextQoL.cs:40]: Sure

File: mod/NomaiTextQoL.cs (hunk 4 above)
Comment (GameWyrm) [reply mod/NomaiTextQoL.cs:86]: Alright

File: mod/default-config.json (hunk 2 above)
Comment (GameWyrm) [reply mod/default-config.json]: Alright it was tedious but I reverted the indentation

File: mod/APRandomizer.cs (hunk 3 above)
Comment (GameWyrm) [reply mod/APRandomizer.cs:338]: Going to leave this as is unless you really don't like it being here.

File: mod/NomaiTextQoL.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (GameWyrm) [reply mod/NomaiTextQoL.cs:39]: Done

File: mod/NomaiTextQoL.cs (hunk 4 above)
Comment (GameWyrm) [reply mod/NomaiTextQoL.cs:89]: Done

**Pull request merged**